services/ad-server/src/public/js/decision-logic.js

```diff
@@ -0,0 +1,39 @@
+/*
+ Copyright 2022 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+function scoreAd(
+  adMetadata,
+  bid,
+  auctionConfig,
+  trustedScoringSignals,
+  browserSignals,
+) {
+  const {bidFloor} = auctionConfig.sellerSignals;
+
+  return {
+    desirability: bid > bidFloor ? 1 : 0,
```

Comment:
desirability should be the bid when it's higher than the floor. That way, the auction will be a first-price auction. Right now, if there are multiple bids--all higher than the floor--all of them will have the same desirability score, and the browser will pick a random one as the winner.

Reply:
I see that the next pull request fixes this; feel free to ack and resolve.

Reply:
That code should be in this PR.  I've moved it over.